leshan-core/src/test/java/org/eclipse/leshan/core/request/execute/ArgumentsParsingTest.java

```diff
@@ -0,0 +1,131 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Orange.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ *
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v20.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ *
+ * Contributors:
+ *     Orange - Add better support for Arguments of Execute Operation.
+ *******************************************************************************/
+package org.eclipse.leshan.core.request.execute;
+
+import static org.junit.Assert.*;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import org.junit.Test;
+
+public class ArgumentsParsingTest {
+
+    @Test
+    public void should_parse_text_into_arguments() throws InvalidArgumentException {
+        Arguments arguments = Arguments.parse("3='stringValue'");
+
+        assertNotNull(arguments);
+        assertEquals(1, arguments.size());
+
+        List<Argument> caughtArguments = new ArrayList<>();
+        for (Argument arg : arguments) {
+            caughtArguments.add(arg);
+        }
+
+        assertEquals(1, caughtArguments.size());
+        assertEquals(new Argument(3, "stringValue"), caughtArguments.get(0));
+        assertEquals(3, caughtArguments.get(0).getDigit());
+        assertEquals("stringValue", caughtArguments.get(0).getValue());
```

Comment:
You need to create a list for most of the tests.

This makes me feel that maybe the API need to be improve to avoid that ?
We already have a size()
We maybe also need to getter by digit or index. (I guess digit is more appropriate)

Reply:
All the complications are because I assumed that arguments are list, not map. I'm afraid adding ```getByDigit()``` (or something like that) method in case of more items at same digit. Or maybe I just overthinking?

And what about ```getFirstByDigit()```? We can implement desired functionality and if we establish that we allow only one digit per value (it would be unique), then we can rename method to e.g. ```getByDigit()``` (or just ```get()```)

Reply:
Yep this is related to : https://github.com/eclipse/leshan/pull/1148#discussion_r758270666